File: rock_paper_scissors_backend/src/player_match_model.ts
import MoveModel from "./move_model";

class PlayerMatchModel {
  playerId: string;
  moves: MoveModel[];
  score: number;

  constructor(playerId: string) {
    this.playerId = playerId;
    this.moves = [];
    this.score = 0;
  }
}

export default PlayerMatchModel;

File: rock_paper_scissors_frontend/lib/models/player_match_model.dart
import 'models.dart';

class PlayerMatchModel {
  final String id;
  final List<MoveModel> moves;
  final int score;

  PlayerMatchModel({
    required this.id,
    this.moves = const [],
    this.score = 0,
  });

  factory PlayerMatchModel.fromJson(Map<String, dynamic> json) {
    return PlayerMatchModel(
      id: json['playerId'],
      moves: json['moves'] != null ? (json['moves'] as List).map((json) => MoveModel.fromJson(json)).toList() : [],
      score: json['score'],
    );
  }
}
